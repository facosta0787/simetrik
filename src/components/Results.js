import React from "react";
import "./Results.scss";

function Results({ title, headers, fields, data }) {
  console.log(data);
  return (
    <article className="results">
      <h3 className="results-title">{title}</h3>
      <table className="results-table">
        <thead className="table-head">
          <tr className="table-row">
            {headers.map((header) => (
              <th className="table-header">{header}</th>
            ))}
          </tr>
        </thead>

        <tbody className="table-body">
          {data.map((row) => (
            <tr className="table-row">
              {fields.map((item, idx) => (
                <td data-label={headers[idx]} className="table-cell">
                  {row[item]}
                </td>
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </article>
  );
}

export default Results;
