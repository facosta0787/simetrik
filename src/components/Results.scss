.results {
  &-title {
    margin: 1rem 0 0 0;
    padding: 10px 10px 0 10px;
  }

  .results-table {
    table-layout: fixed;
    width: 100%;
    border-collapse: collapse;

    .table-head {
      display: none;
    }

    .table-body,
    .table-row,
    .table-cell {
      display: block;
    }

    .table-row {
      margin-top: 1rem;
    }

    .table-cell {
      position: relative;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
      border-bottom: 1px solid var(--grey-dark);
    }

    .table-cell::before {
      content: attr(data-label);
      display: inline-block;
      position: relative;
      top: 0;
      width: 29%;
      font-weight: bold;
    }

    @media (min-width: 760px) {
      margin-top: 1rem;
      border-collapse: separate;

      .table-head {
        display: table-header-group;
        background-color: var(--blue);
      }

      .table-header {
        padding: 12px 10px;
        color: var(--white);
      }

      .table-body {
        display: table-row-group;
      }

      .table-row {
        display: table-row;
      }

      .table-cell {
        display: table-cell;
        padding: 12px 10px;
      }

      .table-cell::before {
        display: none;
      }
    }
  }
}
