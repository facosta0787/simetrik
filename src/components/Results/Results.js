import React from 'react'
import './Results.scss'

function Results({ title, headers, fields, data }) {
  return (
    <article data-testid="resutls-article" className="results">
      <h3 className="results-title">{title}</h3>
      <table className="results-table">
        <thead className="table-head">
          <tr className="table-row">
            {headers.map((header, idx) => (
              <th key={`${header}-${idx}`} className="table-header">
                {header}
              </th>
            ))}
          </tr>
        </thead>

        <tbody className="table-body">
          {data.map(row => (
            <tr key={row.id} className="table-row">
              {fields.map((item, idx) => (
                <td
                  key={`${row.id}-${idx}`}
                  data-label={headers[idx]}
                  className="table-cell"
                >
                  {row[item]}
                </td>
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </article>
  )
}

export default Results
