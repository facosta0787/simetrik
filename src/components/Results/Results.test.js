import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Results from './Results'

describe('Restuls Component', () => {
  afterEach(cleanup)

  it('Should match to snapshot', () => {
    const testProps = {
      title: 'this is a title',
      headers: ['id', 'name', 'address'],
      fields: ['id', 'name', 'address'],
      data: [
        {
          id: 1,
          name: 'Tony Stark',
          address: 'Stark Tower',
        },
      ],
    }

    const { container } = render(<Results {...testProps} />)
    expect(container).toMatchSnapshot()
  })
})
