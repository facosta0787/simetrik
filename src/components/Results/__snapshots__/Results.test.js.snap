// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restuls Component Should match to snapshot 1`] = `
<div>
  <article
    class="results"
    data-testid="resutls-article"
  >
    <h3
      class="results-title"
    >
      this is a title
    </h3>
    <table
      class="results-table"
    >
      <thead
        class="table-head"
      >
        <tr
          class="table-row"
        >
          <th
            class="table-header"
          >
            id
          </th>
          <th
            class="table-header"
          >
            name
          </th>
          <th
            class="table-header"
          >
            address
          </th>
        </tr>
      </thead>
      <tbody
        class="table-body"
      >
        <tr
          class="table-row"
        >
          <td
            class="table-cell"
            data-label="id"
          >
            1
          </td>
          <td
            class="table-cell"
            data-label="name"
          >
            Tony Stark
          </td>
          <td
            class="table-cell"
            data-label="address"
          >
            Stark Tower
          </td>
        </tr>
      </tbody>
    </table>
  </article>
</div>
`;
